tests/platform_test.js

```diff
@@ -0,0 +1,175 @@
+      I.seeAttributesOnElements(locate('a').last(), { href: links.privacyPolicy });
+      I.seeTextEquals('Privacy Policy', locate('a').last());
+    });
+
+    // Verify Sign Up button is disabled and Back to Login button is present and enabled
+    I.seeAttributesOnElements(buttons.signUp, { disabled: true });
+    I.seeTextEquals('Back to login', buttons.backToSignIn);
+    I.seeAttributesOnElements(buttons.backToSignIn, { disabled: null });
+
+    // Focus on Email, First Name, Last Name fields and terms checkbox to verify that fields are required
+    I.usePlaywrightTo('focus on email and password fields', async ({ page }) => {
+      page.focus('[data-qa="email-text-input"]');
+      page.focus('[data-qa="firstName-text-input"]');
+      page.focus('[data-qa="lastName-text-input"]');
+      page.focus('[data-qa="agreement-checkbox-input"]');
+    });
+
+    pmmSettingsPage.perconaPlatform.verifyEmailFieldValidation();
+
+    // First Name validation
+    I.seeTextEquals(messages.requiredField, elements.firstNameValidation);
+
+    // Verify there is no validation for not empty field
+    I.appendField(fields.firstNameField, 'First');
+    I.seeTextEquals('', elements.firstNameValidation);
+
+    // Last Name validation
+    I.seeTextEquals(messages.requiredField, elements.lastNameValidation);
+
+    // Verify there is no validation for not empty field
+    I.appendField(fields.lastNameField, 'Last');
+    I.seeTextEquals('', elements.lastNameValidation);
+
+    // Terms checkbox validation
+    I.seeTextEquals(messages.requiredField, elements.termsCheckboxValidation);
+
+    // Verify there is no validation message if checkbox is checked and Sign Up button is enabled
+    I.seeAttributesOnElements(buttons.signUp, { disabled: true });
+    I.forceClick(buttons.termsCheckBox);
+    I.seeAttributesOnElements(buttons.signUp, { disabled: null });
+  },
+);
+
+Scenario(
+  'PMM-T413 Verify user is able to Login @platform @settings',
+  async () => {
+    // Sign In and verify it was successful
+    await pmmSettingsPage.perconaPlatform.login(email, password);
+  },
+);
+
+Scenario(
+  'PMM-T416 Verify user is able to Sign Out @platform @settings',
+  async ({ I, platformAPI }) => {
+    await platformAPI.signIn();
+
+    // Verify user is logged in
+    I.refreshPage();
+    await pmmSettingsPage.perconaPlatform.verifyUserIsLoggedIn(email);
+
+    // Click Sign Out and verify Pop Up message
+    I.click(buttons.signOut);
+    I.verifyPopUpMessage(messages.signedOut);
+
+    // Verify after refresh user still see login form
+    I.waitForVisible(elements.signInForm);
+    I.refreshPage();
+    I.waitForVisible(elements.signInForm);
+  },
+);
+
+Scenario(
+  'PMM-T399 Verify user is able to Sign Up @platform @settings',
+  async ({ I }) => {
+    I.click(buttons.goToSignUp);
+    I.waitForVisible(elements.signUpForm, 30);
+
+    const newUserEmail = await I.generateNewEmail();
+    const newUserPassword = 'MySuperSecretTempPassword123321';
```

Comment:
could use faker too here?

Reply:
faker has a bug actually. I can use another library for generating a password. more details [here](https://github.com/percona-platform/saas-ui/blob/main/cypress/pages/auth/getUser.ts#L4-L7)